app/routes.py

```
@@ -620,389 +445,275 @@ def historial_requisiciones():
+    opciones_estado_permitidas = requisicion_service.get_opciones_estado_permitidas(requisicion, rol_actual_nombre)
 
     form_estado.estado.choices = opciones_estado_permitidas
-    if not form_estado.estado.choices:
+    if not form_estado.estado.choices: # Salvaguarda por si el servicio devuelve lista vacía inesperadamente
         form_estado.estado.choices = [('N/A', 'No disponible')]
 
-    if request.method == 'POST' and form_estado.submit_estado.data and form_estado.validate_on_submit() :
-        if not (current_user.rol_asignado and current_user.rol_asignado.nombre in ['Admin', 'Compras', 'Almacen']):
+
+    # Procesar cambio de estado si el formulario es enviado
+    if form_estado.validate_on_submit() and form_estado.submit_estado.data :
```

Flake8 (7.2.0)

[error] whitespace before ':'

(E203)